sections/main-search.liquid

```diff
@@ -59,152 +59,152 @@
+                  {%- else -%}
+                    {{ 'products.facets.product_count' | t: product_count: search.products_count, count: search.all_products_count }}
+                  {%- endif -%}
+                </span>
+              </h2>
+              <div class="loading-overlay__spinner">
+                <svg aria-hidden="true" focusable="false" class="spinner" viewBox="0 0 66 66" xmlns="http://www.w3.org/2000/svg">
+                  <circle class="path" fill="none" stroke-width="6" cx="33" cy="33" r="30"></circle>
+                </svg>
+              </div>
             </div>
-          </div>
-        </form>
-      </facet-filters-form>
-    {%- endif -%}
-    <div{% if section.settings.filter_type == 'vertical' %} class="facets-vertical page-width"{% endif %}>
-      {%- if search.filters != empty -%}
-        {%- if section.settings.enable_filtering or section.settings.enable_sorting -%}
-          <aside aria-labelledby="verticalTitle" class="facets-wrapper{% unless section.settings.enable_filtering %} facets-wrapper--no-filters{% endunless %}{% if section.settings.filter_type != 'vertical' %} page-width{% endif %}" id="main-search-filters" data-id="{{ section.id }}">
-            {% render 'facets', results: search, enable_filtering: section.settings.enable_filtering, enable_sorting: section.settings.enable_sorting, filter_type: section.settings.filter_type %}
-          </aside>
-        {%- endif -%}
+          </form>
+        </facet-filters-form>
       {%- endif -%}
-      <div class="product-grid-container" id="ProductGridContainer">
-        {%- if search.results.size == 0 and search.filters != empty -%}
-          <div class="template-search__results collection collection--empty{% if section.settings.filter_type != 'vertical' %} page-width{% endif %}" id="product-grid" data-id="{{ section.id }}">
-            <div class="loading-overlay gradient"></div>
-            <div class="title-wrapper center">
-              <h2 class="title title--primary">
-                {{ 'sections.collection_template.empty' | t }}<br>
-                {{ 'sections.collection_template.use_fewer_filters_html' | t: link: search_url, class: "underlined-link link" }}
-              </h2>
+      <div{% if section.settings.filter_type == 'vertical' %} class="facets-vertical page-width"{% endif %}>
+        {%- if search.filters != empty -%}
+          {%- if section.settings.enable_filtering or section.settings.enable_sorting -%}
+            <aside aria-labelledby="verticalTitle" class="facets-wrapper{% unless section.settings.enable_filtering %} facets-wrapper--no-filters{% endunless %}{% if section.settings.filter_type != 'vertical' %} page-width{% endif %}" id="main-search-filters" data-id="{{ section.id }}">
+              {% render 'facets', results: search, enable_filtering: section.settings.enable_filtering, enable_sorting: section.settings.enable_sorting, filter_type: section.settings.filter_type, paginate: paginate %}
```

Comment:
Is it needed here to pass in the `paginate` object here 🤔 I don't see it referenced in the snippet. 
If we an attribute to the `render` we want to add it to the snippet itself in the description at the top of the file: 
<details>
<summary>Screenshot</summary>

![](https://screenshot.click/17-38-6mxip-sh5xw.png)
</details>

Reply:
`result.results_count` benefits from having a `paginate` context within the `facets` render to reduce unnecessary data retrieval. Even though the `paginate` block wraps where the render in `main-search.liquid` and `main-collection-product-grid.liquid` it is not in the context inside of the rendered liquid partial by default.

https://shopify.dev/docs/api/liquid/tags/render

> Inside snippets and app blocks, you can't directly access variables that are [created](https://shopify.dev/docs/api/liquid/tags/variable-tags) outside of the snippet or app block. However, you can [specify variables as parameters](https://shopify.dev/docs/api/liquid/tags/render#render-passing-variables-to-a-snippet) to pass outside variables to snippets.